include/awkward/Reducer.h

```diff
@@ -679,6 +766,10 @@ namespace awkward {
   /// ReducerMin and ReducerMax are exceptions to the no-state rule: they are
   /// parameterized by `initial`, used as an initial value in the minimization
   /// or maximization.
+  ///
+  /// For the complex numbers array it
+  /// returns the minimumm absolute value (also known as norm, modulus,
+  /// or magnitude) of a complex number.
```

Comment:
@sjperkins - as implemented in this PR, the reduces `min`, `max`, `argmin`, `argmax` will return either min/max absolute value or its index. You are welcome to comment or change the code to anything more suitable.

Reply:
Thanks for pinging me @ianna! I guess that it would be most suitable if the behaviour matches numpy's for these functions?

Reply:
Thanks @sjperkins ! `numpy` implements `min`, `max` based on `sort`. The sort order for complex numbers is lexicographic:
https://numpy.org/doc/stable/reference/generated/numpy.sort.html 
I'd better disable these reducers to avoid confusion.

@jpivarski - should this functionality be provided as the `absmin`, `absmax` reducers?

Reply:
Does NumPy really define an ordering on complex numbers?

If users want to sort complex numbers (or find their minimum/maximum) by absolute value or some other criteria, they can construct an intermediate. Calling Python's `abs` on the Awkward Array propagates `np.absolute` to every NumpyArray in the structure, so if the `dtype` is properly preserved as complex, `abs` would automatically give the modulus of all the complex numbers in an array. Then, users can argsort (or argmin/argmax) and apply it the result to the original array. We don't have to provide every option when they can be constructed.

Reply:
Thanks, @jpivarski ! I'll remove `min`, `max`, `argmin`, `argmax` complex number reducers.

Reply:
> Does NumPy really define an ordering on complex numbers?
> 
strangely, yes, a lexicographic one:
```python
>>> import numpy as np
>>> zarray = np.array([[1+2j,1+3j],[5+6j,3+8j]])
>>> zarray
array([[1.+2.j, 1.+3.j],
       [5.+6.j, 3.+8.j]])
>>> zarray.max()
(5+6j)
>>> np.sort(zarray)
array([[1.+2.j, 1.+3.j],
       [3.+8.j, 5.+6.j]])

```

Reply:
In that case, I guess that's what we have to do for compatibility. Do you already have that lexicographic sorting implemented for complex numbers? If so, then that's what should be included.

If you don't already have it implemented, it would be fair to `throw std::runtime_error("sorting complex numbers not implemented yet")`.